src/main/daml/Daml/Finance/Claims/Lifecycle/Rule.daml

```diff
@@ -60,7 +60,7 @@ template Rule
         else do
           let
             currentKey = BaseInstrument.getKey $ toInterface claimInstrument
-            newKey = currentKey with version = sha256 $ show remaining
+            newKey = currentKey with version = sha256 $ mconcat [show v.eventTime, show remaining]
```

Comment:
you could do
newKey = currentKey with version = sha256 . mconcat $ show <$> [v.eventTime, remaining]